lib/foreman_tasks/cleaner.rb

```diff
@@ -138,8 +135,49 @@ def with_backup_file(backup_dir, file_name)
     end
 
     def delete_dynflow_plans(chunk)
-      dynflow_ids = chunk.find_all { |task| task.is_a? Task::DynflowTask }.map(&:external_id)
-      ForemanTasks.dynflow.world.persistence.delete_execution_plans({ 'uuid' => dynflow_ids }, batch_size, @backup_dir)
+      delete_dynflow_plans_by_uuid chunk.find_all { |task| task.is_a? Task::DynflowTask }.map(&:external_id)
+    end
+
+    def delete_dynflow_plans_by_uuid(uuids)
+      ForemanTasks.dynflow.world.persistence.delete_execution_plans({ 'uuid' => uuids }, batch_size, @backup_dir)
+    end
+
+    def delete_orphaned_dynflow_tasks
+      with_noop(orphaned_dynflow_tasks, 'orphaned execution plan') do |source, name|
+        with_batches(source, name) do |chunk|
+          delete_dynflow_plans_by_uuid chunk.select_map(:uuid)
+        end
+      end
+    end
+
+    # source must respond to :count and :limit
+    def with_noop(source, name, noop_message = nil)
+      if noop
+        say '[noop] ' + noop_message if noop_message
+        say "[noop] #{source.count} #{name}(s) would be deleted"
+      else
+        yield source, name
+      end
+    end
+
+    def with_batches(source, name)
+      count = source.count
+      if count.zero?
+        say("No #{name}s found, skipping.")
```

Comment:
Passing the arguments as plural + not having to use the additional `s` would be probably a bit more flexible. When I was testing, I got message:

```
No tasks found, skipping
```

It was not clear what tasks those were. Having `name` something like `tasks matching filter` would be more self-describing.